### PR TITLE
Opensrp-scheduling refactored and redesigned

### PR DESCRIPTION
**Pull request by maimoonak**
Opensrp-scheduling refactored and redesigned as explained in https://github.com/OpenSRP/opensrp-server/issues/48.

### CODE REVIEW

File: opensrp-connector/src/test/java/EncounterTest.java [hunk 1]
```diff
@@ -45,45 +43,35 @@ public void setup() throws IOException{
...
-    			+ " \"formName\": \"pnc_1st_registration\",  \"instanceId\": \"759b4790-47eb-457f-a5b9-3015bc89253c\"  }";
-	
-		FormSubmission fs = new Gson().fromJson(fsStr, FormSubmission.class);
-
-		System.out.println(oc.isOpenmrsForm(fs));
+	public void testEncounter() throws JSONException, ParseException, IOException {
```
Comment (ndisha) [inline opensrp-connector/src/test/java/EncounterTest.java]: @maimoonak what is being tested here? I was expecting 'an assert' or equivalent..

File: opensrp-core/src/main/java/org/opensrp/service/reporting/ChildReportingService.java [hunk 2]
```diff
@@ -1,51 +1,146 @@
...
+import static ch.lambdaj.Lambda.extract;
+import static ch.lambdaj.Lambda.having;
+import static ch.lambdaj.Lambda.on;
+import static ch.lambdaj.Lambda.selectFirst;
 import static java.util.Arrays.asList;
 import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.hamcrest.Matchers.equalTo;
+import static org.joda.time.LocalDate.parse;
 import static org.opensrp.common.AllConstants.ANCFormFields.REGISTRATION_DATE;
-import static org.opensrp.common.AllConstants.ChildCloseFormFields.*;
```
Comment (ndisha) [inline opensrp-core/src/main/java/org/opensrp/service/reporting/ChildReportingService.java]: @maimoonak is there a specific reason we now want to list all the fields vis a vi having them as .*? Just wondering.

File: opensrp-connector/src/test/java/EncounterTest.java (hunk 1 above)
Comment (maimoonak) [inline opensrp-connector/src/test/java/EncounterTest.java]: not complete yet, would be completed after we get HH or any register finalized

File: opensrp-core/src/main/java/org/opensrp/service/reporting/ChildReportingService.java (hunk 2 above)
Comment (maimoonak) [inline opensrp-core/src/main/java/org/opensrp/service/reporting/ChildReportingService.java]: auto organize imports does this...

**Comment by maimoonak** [comment]
merging pull request into master

**Pull request merged**